recipes/proposal/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.32.0"
+
+
+class PROPOSALConan(ConanFile):
+    name = "proposal"
+    homepage = "https://github.com/tudo-astroparticlephysics/PROPOSAL"
+    license = "LGPL-3.0"
+    exports_sources = "CMakeLists.txt"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "monte Carlo based lepton and photon propagator"
+    topics = ("propagator", "lepton", "photon", "stochastic")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_python": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_python": False,
+    }
+
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("cubicinterpolation/0.1.4")
+        self.requires("spdlog/1.8.2")
+        if self.options.with_python:
+            self.requires("pybind11/2.6.2")
```

Comment:
nlohmann_json is vendored, could it be removed and added as external requirement?

Reply:
Is there a way to explicitly set the variable in the recipe to true as long as the PR isn't merged?

Reply:
What you can do in PROPOSAL is to add an option in CMake, to select either vendored nlohmann_json or external one, but it might require significant work.

Oh, I noticed an issue in PROPOSAL with recent versions of nlohmann_json, here is a fix:
https://github.com/SpaceIm/conan-proposal/blob/testing/7.0.1/patches/0005-fix-functional-style-cast.patch

Reply:
I think I have already done the relevant things in this [commit](https://github.com/tudo-astroparticlephysics/PROPOSAL/commit/7cd5e10b26b044a7bf63795738f6b9a5894bec68). :sweat_smile: 

But I have to set `multiple_headers` explicit to true to use `json_fwd.hpp` as long as the PR is not merged, right?

The question is how I can set this variable explicitly in conanfile.py?

Reply:
You can force this value in `configure()`: `self.options["nlohmann_json"].multiple_headers = True`, then raise in `validate()` if value is not honored. But for the moment, it won't help for this PR, because CI of conan-center-index doesn't call conan with `-b missing`.

Honestly, it would be better to remove this option from `nlohmann_json` recipe, and always package multiple headers as I said in https://github.com/conan-io/conan-center-index/pull/5188.